### ISSUE TITLE
Name conflict in generated types

### ISSUE THREAD

**Issue by bradleyayers** [SweetIQ/schemats#86]
I have a table `document` with a column `type` of type `document_type`:

```sql
CREATE TYPE document_type AS ENUM (
    'NOTE',
    'PAGE'
);
CREATE TABLE document (
    id uuid NOT NULL,
    document_type document_type NOT NULL
);
```

This causes a name conflict in the generate types:

```ts
export type document_type = "NOTE" | "PAGE";

// ...

export namespace documentFields {
  export type id = string;
  export type document_type = document_type;
  //          ^^^^^^^^^^^^^
  // Type alias 'note_type' circularly references itself
}
```

**Comment by abenhamdine** [comment]
To avoid that, I think enums could be generated in PascalCase (after all it's the common practice with typescript) and/or with Enum prefix, something like :  

```ts
export type EnumDocumentType = "NOTE" | "PAGE";

export namespace documentFields {
  export type id = string;
  export type document_type = EnumDocumentType;
}

**Comment by bradleyayers** [comment]
I like that proposal, nice and simple!

**Comment by abenhamdine** [comment]
It will be a breaking change though, because someone could have imported an enum and used it in its code.

**Comment by bradleyayers** [comment]
Perhaps an option/flag then?

**Comment by abenhamdine** [comment]
@xiamx what do you think on it ?
Personnaly, I'm -1 on putting this behind a flag because options complexify config and code.

It seems better to include it in next major version (while I doubt many users have enums and import them).

**Comment by xiamx** [comment]
Yes, preferring a major version bump to have Enum in PascalCase.
We need a minimal impact solution to handle name collision for the current major version.

**Comment by bradleyayers** [comment]
Emitting exports for both snake and camel, but using camel in the interfaces should work

**Comment by xiamx** [comment]
Nice, that should provide a smooth transition.

**Comment by hinderberg** [comment]
Whats the status on this?

**Comment by bradleyayers** [comment]
I'm using a forked version that I have checked into my project with this fix applied.

**Comment by hinderberg** [comment]
Thats to bad! Renaming my enum type while waithing for this. Forking too many libs allready! ;)

**Comment by felixmosh** [comment]
Why not export the enums and all table related interfaces under namespace with the name of the table.
